internal/util/conditions.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type ConditionType string
+
+type ConditionReason string
+
+func CheckCondition(
+	conditions []metav1.Condition,
+	typ ConditionType,
+	reason ConditionReason,
+	status metav1.ConditionStatus,
+	resourceGeneration int64,
+) bool {
+	for _, cond := range conditions {
+		if cond.Type == string(typ) &&
+			cond.Reason == string(reason) &&
+			cond.Status == status &&
+			cond.ObservedGeneration >= resourceGeneration {
```

Comment:
Shouldn't the generation match exactly? It seems we're checking for an equality in the current code. 
Also [this](https://github.com/kubernetes/apimachinery/blob/825264154b159a97fc8d577c677d0c1ce434ca30/pkg/apis/meta/v1/types.go#L1480-L1485) seem to indicate the equality is the way to go. But I might be wrong here.

Reply:
As I understand in practice both `>=` and `==` will work the same as it's not expected to happen that the observed generation will be higher than the actual resource's generation. I will replace it with `==` though to not introduce ambiguity: [8728d24](https://github.com/Kong/kubernetes-ingress-controller/pull/2952/commits/8728d24b2ecef90b0fa64f3ff515226368a14b3c)